descartes_planner/src/planning_graph.cpp

```diff
@@ -814,45 +814,71 @@ bool PlanningGraph::calculateJointSolutions(const std::vector<TrajectoryPtPtr>&
 {
   poses.resize(points.size());
 
-  for (std::size_t i = 0; i < points.size(); ++i)
+  bool success = true;
+  #pragma omp parallel shared(success)
   {
-    std::vector<std::vector<double>> joint_poses;
-    points[i]->getJointPoses(*robot_model_, joint_poses);
-
-    if (joint_poses.empty())
-    {
-      ROS_ERROR_STREAM(__FUNCTION__ << ": IK failed for input trajectory point with ID = " << points[i]->getID());
-      return false;
-    }
-
-    poses[i].reserve(joint_poses.size());
-    for (auto& sol : joint_poses)
+    #pragma omp for
+    for (std::size_t i = 0; i < points.size(); ++i)
     {
-      poses[i].emplace_back(std::move(sol), points[i]->getTiming());
+      #pragma omp flush (success)
+      if (success) // stop all threads after first observed failure
```

Comment:
this should be a for loop break if success == false

Reply:
That was my initial thought (and attempt actually) but OpenMP apparently does not allow breaks in parallel for loops.  My thought was that, if I still have to loop at least don't do anything time consuming...  which led to the above implementation.

Reply:
Also, a failure at this point means you most likely have problems elsewhere.  That said, a few extra for-loop iterations aren't going to matter in the grand scheme of things.

Reply:
Actually... after reading quick, it seems OpenMP 4.0 added a `cancel` directive (see page 113 [here](http://www.openmp.org/wp-content/uploads/OpenMP_Examples_4.0.1.pdf) for example).  I will check if this works.